backend/services/implementations/testSessionService.ts

```diff
@@ -352,7 +353,12 @@ class TestSessionService implements ITestSessionService {
     // - index (for multiple choice)
     // - list of indices (for multiple select)
     // - null (for no answer)
-    const studentTestAnswers: (number[] | number | null)[][] = result.answers;
+    const studentTestAnswers: (
+      | number[]
+      | number
+      | null
+      | FractionMetadata
```

Comment:
i'm not sure it makes sense here to just tack on another type. instead, it might make more sense to convert the `answers` field type in `ResultRequestDTO` from `(number[] | number | null)[][]` to `(number[] | null)` and for short answer / multiple choice responses, just store as an array of size 1. i think then we can combine the `getCorrectAnswer` and `getCorrectAnswers` functions  here for a cleaner implementation (and also keep the typing a bit cleaner)

cc @carissa-tang since you have a bit more context, is there any reason why this wouldn't work? i also think should be a minor change, but lmk if i'm wrong about that and if we should push that work to another PR.
ik you also have a ticket coming up to refactor the test session grading, so lmk if it makes more sense to do it there.